src/allmydata/client.py

```diff
@@ -561,6 +576,34 @@ def tub_creator(handler_overrides=None, **kwargs):
     return sb
 
 
+def _load_grid_manager_certificates(config):
+    """
+    Load all Grid Manager certificates in the config in a list. An
+    empty list is returned if there are none.
+    """
+    grid_manager_certificates = []
+
+    cert_fnames = list(config.enumerate_section("grid_manager_certificates").values())
+    for fname in cert_fnames:
+        fname = config.get_config_path(fname.decode('utf8'))
+        if not os.path.exists(fname):
+            raise ValueError(
+                "Grid Manager certificate file '{}' doesn't exist".format(
+                    fname
+                )
+            )
+        with open(fname, 'r') as f:
+            cert = json.load(f)
+        if set(cert.keys()) != {"certificate", "signature"}:
+            raise ValueError(
+                "Unknown key in Grid Manager certificate '{}'".format(
+                    fname
+                )
+            )
+        grid_manager_certificates.append(cert)
+    return grid_manager_certificates
```

Comment:
In #897 I added a method to `_Config` that is conceptually similar to this one *instead* of making a free function that operates on a config object.  Initially I wanted to make it a free function because that makes it more general and re-usable but I thought that "poking paths in the config hierarchy" meant it's not actually very general so it should just be a `_Config` method.

If it's a `_Config` method then an alternate future `_Config` implementation (eg one based on SQLite3) can still implement it correctly.  But the free function will still want to poke the filesystem.

Reply:
...so the point here being "(more) random code calling `_Config.get_config_path()` is bad" basically, right?

Reply:
I *think* so.  In our extensive free time we might try to start an alternate implementation of the (implicit) `_Config` interface and see where we run into problems.  Having a second implementation is a good way to bring issues like this into focus.